### PR TITLE
test: Add missing trace filter rules test

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request:

- Removed not more required grafan service filter rules
- Add new filter rule for metric gateway traces
- Add e2e tests for filter rules

## Related Issues and Documents

Closes:

Related issues:
https://github.com/kyma-project/telemetry-manager/issues/410
## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: test/e2e/traces_noisy_span_filter_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	kittraces "github.com/kyma-project/telemetry-manager/test/testkit/otlp/traces"
+	"github.com/kyma-project/telemetry-manager/test/testkit/verifiers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kyma-project/telemetry-manager/test/testkit/periodic"
+)
+
+var _ = Describe("Traces Noisy Span Filter", Label("tracing"), func() {
```
Comment (chrkl) [inline test/e2e/traces_noisy_span_filter_test.go:26]: Let's make this description a proper sentence. Maybe "Filter Noisy Trace Spans"

File: test/testkit/verifiers/traces.go [hunk 2]
```diff
@@ -71,3 +71,18 @@ func TracesShouldBeDelivered(proxyClient *apiserver.ProxyClient, telemetryExport
 		g.Expect(err).NotTo(gomega.HaveOccurred())
 	}, periodic.EventuallyTimeout, periodic.TelemetryInterval).Should(gomega.Succeed())
 }
+
+func TracesShouldNotBePresent(proxyClient *apiserver.ProxyClient, telemetryExportURL string, traceID pcommon.TraceID, spanIDs []pcommon.SpanID, attrs pcommon.Map, resAttrs pcommon.Map) {
+	gomega.Eventually(func(g gomega.Gomega) {
```
Comment (chrkl) [inline test/testkit/verifiers/traces.go:76]: I think we should use `Consistently` for this check. This implementation has the risk that we already succeed before anything arrives in the mock backend.

File: test/e2e/traces_noisy_span_filter_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		It("Should have a running pipeline", Label(operationalTest), func() {
+			verifiers.TracePipelineShouldBeRunning(ctx, k8sClient, pipelineName)
+		})
+
+		It("Should verify end-to-end trace delivery", Label(operationalTest), func() {
+			traceID, spanIDs, attrs := kittraces.MakeAndSendTraces(proxyClient, urls.OTLPPush())
+			verifiers.TracesShouldBeDelivered(proxyClient, urls.MockBackendExport(mockBackendName), traceID, spanIDs, attrs)
+		})
+
+		It("Should filter noisy traces and spans", Label(operationalTest), func() {
```
Comment (chrkl) [inline test/e2e/traces_noisy_span_filter_test.go:95]: Maybe let's split this to one `It` per test case to make it better readable. Maybe use the `By` for the step of sending traces.

File: test/e2e/traces_noisy_span_filter_test.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		BeforeAll(func() {
+			k8sObjects := makeResources()
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should be able to get trace gateway metrics endpoint", Label(operationalTest), func() {
```
Comment (chrkl) [inline test/e2e/traces_noisy_span_filter_test.go:78]: I would rather not add this to the operational / upgrade test set.

File: test/e2e/traces_noisy_span_filter_test.go (hunk 3 above)
Comment (hisarbalik) [reply test/e2e/traces_noisy_span_filter_test.go:95]: Discussion with Stas abut this point, we come to the conclusion, doing dedicated `It` per test case will put more pressure on the API Server

Comment (chrkl) [reply test/e2e/traces_noisy_span_filter_test.go:95]: I don't understand that point. It is only about the organization of the test cases. The API server calls should not change.

File: test/testkit/verifiers/traces.go (hunk 2 above)
Comment (hisarbalik) [reply test/testkit/verifiers/traces.go:76]: changed

File: test/e2e/traces_noisy_span_filter_test.go (hunk 1 above)
Comment (hisarbalik) [reply test/e2e/traces_noisy_span_filter_test.go]: Updated

File: test/e2e/traces_noisy_span_filter_test.go (hunk 4 above)
Comment (hisarbalik) [reply test/e2e/traces_noisy_span_filter_test.go]: label removed

File: test/testkit/verifiers/traces.go (hunk 2 above)
Comment (chrkl) [reply test/testkit/verifiers/traces.go:76]: It's still an `Eventually` block. Maybe you forgot to push this?

File: test/e2e/traces_noisy_span_filter_test.go (hunk 3 above)
Comment (hisarbalik) [reply test/e2e/traces_noisy_span_filter_test.go]: each test case moved in own execution block

File: test/e2e/traces_noisy_span_filter_test.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+			WithOutputEndpointFromSecret(mockBackend.HostSecretRef()).
+			Persistent(isOperational())
+		pipelineName = pipeline.Name()
+		objs = append(objs, pipeline.K8sObject())
+
+		urls.SetOTLPPush(proxyClient.ProxyURLForService(
+			kitkyma.SystemNamespaceName, "telemetry-otlp-traces", "v1/traces/", ports.OTLPHTTP),
+		)
+
+		traceGatewayExternalService := kitk8s.NewService("telemetry-otlp-traces-external", kitkyma.SystemNamespaceName).
```
Comment (skhalash) [inline test/e2e/traces_noisy_span_filter_test.go:57]: We don't need an external service anymore, we can call ClusterIP services using the proxyClient

**Review by skhalash — CHANGES_REQUESTED** [review]

File: test/e2e/traces_noisy_span_filter_test.go (hunk 5 above)
Comment (hisarbalik) [reply test/e2e/traces_noisy_span_filter_test.go:57]: removed

**Review by chrkl — APPROVED** [review]

**Review by skhalash — DISMISSED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**